src/Services/ClientRegistry.php

```diff
@@ -64,6 +69,16 @@ private function buildClientConfiguration(array $conf): ClientConfiguration
             $conf['uri'] = self::buildConnectionUri($conf['hosts']);
         }
 
+        $conf['uriOptions'] = [
+            'replicaSet' => $conf['replicaSet'],
+            'ssl' => $conf['ssl'],
+            'connectTimeoutMS' => $conf['connectTimeoutMS'],
+            'readPreference' => $conf['readPreference'],
+        ];
+        if ($this->uriOptionsService instanceof UriOptionsInterface) {
+            $conf['options'] = $this->uriOptionsService->buildUriOptions($conf['uriOptions']);
+        }
+
```

Comment:
Since we have those defaults, we should make clear in the documentation that defining the URI options service, the other config fields won't be taken into account. 

What is the effect of having them null? 
Should be merge the default with the result of the URI options factory instead?
In this scenario, we will take into account the configuration fields,  allowing them to be overwritten by the URI options factory.

Reply:
Users can use the URI options service to both override the default configuration fields or add new ones

Reply:
Ok, so

- if is not defined any UriOptionInterface service -> the bundle should behave as it is by now
- if a UriOptionInterface service -> the behaviour depends on `buildUriOptions`

LGTM